akka-distributed-data/src/main/scala/akka/cluster/ddata/Replicator.scala

```diff
@@ -1111,15 +1117,22 @@ final class Replicator(settings: ReplicatorSettings) extends Actor with ActorLog
     }
   }
 
-  def receiveMemberUp(m: Member): Unit =
+  def receiveWeaklyUpMemberUp(m: Member): Unit =
     if (matchingRole(m) && m.address != selfAddress)
+      weaklyUpNodes += m.address
```

Comment:
I vaguely remember `union` be faster than `+=` though not sure if that's the case still.
Does not matter that much here though, so don't worry about it :)